solr/modules/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java

```diff
@@ -241,10 +242,31 @@ public void setContext(ResultContext context) {
      *
      * @param transformerFeatureStore the explicit transformer feature store
      */
-    private LoggingModel createLoggingModel(String transformerFeatureStore) {
-      final ManagedFeatureStore fr = ManagedFeatureStore.getManagedFeatureStore(req.getCore());
+    private LoggingModel createLoggingModel(
+        String transformerFeatureStore, Boolean docsWereNotReranked) {
+      final ManagedFeatureStore featureStores =
+          ManagedFeatureStore.getManagedFeatureStore(req.getCore());
+
+      // check for non-existent feature store name
+      if (transformerFeatureStore != null) {
+        if (!featureStores.getStores().containsKey(transformerFeatureStore)) {
+          throw new SolrException(
+              SolrException.ErrorCode.BAD_REQUEST,
+              "Missing feature store: " + transformerFeatureStore);
+        }
+      }
```

Comment:
let me rephrase it:
what happens if you entirely remove this block
`// check for non-existent feature store name
       if (transformerFeatureStore != null) {
         if (!featureStores.getStores().containsKey(transformerFeatureStore)) {
           throw new SolrException(
               SolrException.ErrorCode.BAD_REQUEST,
               "Missing feature store: " + transformerFeatureStore);
         }
       }`
       
       Shouldn't 'final FeatureStore store = featureStores.getFeatureStore(transformerFeatureStore);' raise the same exception?

Reply:
I added the exception in the _doGet_ method, so _getFeatureStore()_ remains unchanged and doesn't raise any exception as it is also used to put new feature stores.